### ISSUE TITLE
Robolectric Service Testing

### ISSUE THREAD

**Issue by goldy1992** [google/dagger#1969]
Given a Service class such as:

```
@AndroidEntryPoint
class MyService() : Service {
  // example dependency
  @Inject
  lateinit var myVariable : String

   // class code
}
```

When we come to write the Robolectric test e.g.

```
@RunWith(RobolectricTestRunner::class)
@UninstallModules(...)
@Config(application = HiltTestApplication::class)
@HiltAndroidTest
class MediaPlaybackServiceTest {

  @Rule @JvmField
  val rule : HiltAndroidRule = HiltAndroidRule(this)

  @Before
  fun setup() {
     rule.inject()
     // more setup
  }

  @Test
  fun myTest() {
    val service : MyService = Robolectric.setupService(MyService::class.java)
    // The rest of the test
  }
```

**Expect**: To be able to generate an instance of `MyService` in order to test
**Actual**:  `kotlin.UninitializedPropertyAccessException: lateinit property myVariable has not been initialized`

It looks like **Robolectric is not using Hilt correctly to set up the service.**

Relevant sections of the build.gradle
```
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-android-extensions'
apply plugin: 'kotlin-kapt'
apply plugin: 'dagger.hilt.android.plugin'

android {
// android code
}

dependencies {
      // hilt
    def HILT_VERSION = '2.28-alpha'
    implementation group: 'com.google.dagger', name: 'hilt-android', version: HILT_VERSION
    kapt group: 'com.google.dagger', name: 'hilt-android-compiler', version: HILT_VERSION

    testImplementation group: 'com.google.dagger', name: 'hilt-android-testing', version: HILT_VERSION
    kaptTest group: 'com.google.dagger', name: 'hilt-android-compiler', version: HILT_VERSION

   def ROBOLECTRIC_VERSION = '4.3.1'
   testImplementation("org.robolectric:robolectric:$ROBOLECTRIC_VERSION") {
        exclude group: "com.google.auto.service", module: "auto-service"
    }
    testImplementation group: 'org.robolectric', name: 'shadows-supportv4', version: ROBOLECTRIC_VERSION
 
  // other depenencies
}
```

**Comment by bcorso** [comment]
Unfortunately, the bytecode injected classes currently don't get picked up on class path when running Robolectric tests via AndroidStudio. We're hoping to fix this once [b/37076369](https://issuetracker.google.com/37076369) is fixed.

In the meantime, see https://github.com/google/dagger/issues/1956 for a couple workarounds.

We also plan to explicitly call this out in the docs. I'm in the process of making those changes now.

**Comment by goldy1992** [comment]
I can confirm that the workaround of using the Gradle test runner works and for me suffices in giving reason to close this issue.

Many thanks @bcorso
